enigma-tools-u/src/web3_utils/enigma_contract.rs

```diff
@@ -23,54 +24,38 @@ pub struct EnigmaContract {
     pub web3: Arc<Web3<Http>>,
     pub eloop: EventLoopHandle,
     pub w3_contract: Contract<Http>,
```

Comment:
is this still necessary here? seems like it's needed only for the address that I'm guessing can be extracted from the ethabi contract as well?

Reply:
Good point, i'll look into it.

Reply:
nope, we also use its `query` method under `enigma-tools-u/web3_utils::enigma_contract::<impl ContractQueries for EnigmaContract>`